From 81d03df6d6debd837ba9bb441bf02e7383fe595f Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Fri, 5 Oct 2018 02:54:14 +0200
Subject: [PATCH 5/5] sdk: Don't clear calling identify when calling
 IOverlayManager.setEnabled()

* Since changing overlays is now guarded with enforceCallingPermission()
  instead of enforceCallingOrSelfPermission() we can no longer call
  setEnabled() after resetting our identify.

Change-Id: I25df30d0ba83f15ca1124d6fc6f79d3465d052f2
---
 .../internal/StyleInterfaceService.java       | 20 ++-----------------
 1 file changed, 2 insertions(+), 18 deletions(-)

diff --git a/lineage/lib/main/java/org/lineageos/platform/internal/StyleInterfaceService.java b/lineage/lib/main/java/org/lineageos/platform/internal/StyleInterfaceService.java
index 9085b5d..edc36ba 100644
--- a/lineage/lib/main/java/org/lineageos/platform/internal/StyleInterfaceService.java
+++ b/lineage/lib/main/java/org/lineageos/platform/internal/StyleInterfaceService.java
@@ -318,15 +318,7 @@ public class StyleInterfaceService extends LineageSystemService {
         @Override
         public boolean setAccent(String pkgName) {
             enforceChangeStylePermission();
-            /*
-             * We need to clear the caller's identity in order to
-             *   allow this method call to modify settings
-             *   not allowed by the caller's permissions.
-             */
-            long token = clearCallingIdentity();
-            boolean success = setAccentInternal(pkgName);
-            restoreCallingIdentity(token);
-            return success;
+            return setAccentInternal(pkgName);
         }
 
         @Override
@@ -384,15 +376,7 @@ public class StyleInterfaceService extends LineageSystemService {
         @Override
         public boolean setDarkOverlay(String overlayName) {
             enforceChangeStylePermission();
-            /*
-             * We need to clear the caller's identity in order to
-             *   allow this method call to modify settings
-             *   not allowed by the caller's permissions.
-             */
-            long token = clearCallingIdentity();
-            boolean success = setDarkOverlayInternal(overlayName);
-            restoreCallingIdentity(token);
-            return success;
+            return setDarkOverlayInternal(overlayName);
         }
 
         @Override
-- 
2.17.1

